src/com/intellij/idea/plugin/hybris/runConfigurations/HybrisJUnitExtension.java

```diff
@@ -61,13 +69,42 @@ public <T extends RunConfigurationBase<?>> void updateJavaParameters(
         if (!params.getEnv().containsKey(HYBRIS_DATA_DIR_ENV)) {
             final HybrisProjectSettings settings = HybrisProjectSettingsComponent.getInstance(project).getState();
 
-            final String hybrisDataDirPath = FileUtil.toCanonicalPath(
-                project.getBasePath() + '/' + settings.getHybrisDirectory() + '/' + HYBRIS_DATA_DIRECTORY);
+            final String hybrisDataDirPath = FileUtil.toCanonicalPath(project.getBasePath() + '/' + settings.getHybrisDirectory() + '/' + HYBRIS_DATA_DIRECTORY);
 
             if (hybrisDataDirPath != null) {
                 params.addEnv(HYBRIS_DATA_DIR_ENV, hybrisDataDirPath);
             }
         }
+        addJdkPropertiesIfNecessary(vmParameters, project);
+    }
+
+    private void addJdkPropertiesIfNecessary(final ParametersList params, final Project project) {
```

Comment:
can you try to rely on `PropertiesService` to get required property?

Reply:
I will change the Implementation to use the `PropertiesService`.

Short question regarding this. In the Service is the method `findAllProperties` which currently not supportes all possibilities to set properties.
First of all the Optionl Config dir can also be set via `HYBRIS_OPT_CONFIG_DIR` Environment Variable (see https://help.sap.com/docs/SAP_COMMERCE/b490bb4e85bc42a7aa09d513d0bcb18e/8beb75da86691014a0229cf991cb67e4.html)

There is also the possibility to define Properties by specifying a File via Environment Property `HYBRIS_RUNTIME_PROPERTIES` . This Property file is loaded before the `HYBRIS_OPT_CONFIG_DIR`

And finally there is the possibility to define Environment Properties for setting Values (see: https://help.sap.com/docs/SAP_COMMERCE/b490bb4e85bc42a7aa09d513d0bcb18e/8b8e13c9866910149d40b151a9196543.html?q=y_#using-environment-variables-instead-of-files-for-secure-settings) 

If desired I can add the Support for those as well so that we will be able to get most of the Properties a User can define.

Reply:
@StefanKruk , this service was introduced by original authors years ago. This service was initially used for ImpEx files as a resolver for `$config-...` properties.

All your comments are valid and it would be great to see them implemented!

p.s. you've missed one very important change, please add yourself to the CONTRIBUTING.md

Reply:
I will first change my current implementation for the Integration Tests as you have suggested and will finalize this PR.
Afterwards I will try to update the Properties Loading. I will most likely create a new PR for this.